improver/developer_tools/metadata_interpreter.py

```diff
@@ -181,7 +182,7 @@ def check_probability_cube_metadata(self, cube: Cube) -> None:
             )
 
         try:
-            self.diagnostic = get_diagnostic_cube_name_from_probability_name(
+            self.diagnostic = get_threshold_coord_name_from_probability_name(
```

Comment:
I agree. This is unnecessarily repetitive. I've changed the code to reuse `self.diagnostic` and to `return` if the exception is raised to avoid using an undefined variable.